library/src/main/scala/scala/scalajs/js/Union.scala

```diff
@@ -63,6 +63,15 @@ object | { // scalastyle:ignore
   implicit def from[A, B1, B2](a: A)(implicit ev: Evidence[A, B1 | B2]): B1 | B2 =
     a.asInstanceOf[B1 | B2]
 
+  /** Upcast `A` to `UndefOr[B1 | B2]`.
+   *
+   * This variant of `from` exists because scala doesnt nest implicit conversions
+   *
+   *  This needs evidence that `A <: B1 | B2`.
+   */
+  implicit def fromU[A, B1, B2](a: A)(implicit ev: Evidence[A, B1 | B2]): UndefOr[B1 | B2] =
+    a.asInstanceOf[UndefOr[B1 | B2]]
```

Comment:
This method should go in the companion object of `js.UndefOr` rather than of `|`, because the "top-level" result type is `js.UndefOr`.

I also suggest you give it a more complete name, such as `any2undefOrUnion`.